src/systems/ackermann_steering/AckermannSteering.cc

```diff
@@ -0,0 +1,771 @@
+    sdfElem = sdfElem->GetNextElement("right_steering_joint");
+  }
+
+  this->dataPtr->wheelSeparation = _sdf->Get<double>("wheel_separation",
+      this->dataPtr->wheelSeparation).first;
+  this->dataPtr->kingpinWidth = _sdf->Get<double>("kingpin_width",
+      this->dataPtr->kingpinWidth).first;
+  this->dataPtr->wheelBase = _sdf->Get<double>("wheel_base",
+      this->dataPtr->wheelBase).first;
+  this->dataPtr->steeringLimit = _sdf->Get<double>("steering_limit",
+      this->dataPtr->steeringLimit).first;
+  this->dataPtr->wheelRadius = _sdf->Get<double>("wheel_radius",
+      this->dataPtr->wheelRadius).first;
+
+  // Parse speed limiter parameters.
+  bool hasVelocityLimits     = false;
+  bool hasAccelerationLimits = false;
+  bool hasJerkLimits         = false;
+  double minVel              = std::numeric_limits<double>::lowest();
+  double maxVel              = std::numeric_limits<double>::max();
+  double minAccel            = std::numeric_limits<double>::lowest();
+  double maxAccel            = std::numeric_limits<double>::max();
+  double minJerk             = std::numeric_limits<double>::lowest();
+  double maxJerk             = std::numeric_limits<double>::max();
+
+  if (_sdf->HasElement("min_velocity"))
+  {
+    minVel = _sdf->Get<double>("min_velocity");
+    hasVelocityLimits = true;
+  }
+  if (_sdf->HasElement("max_velocity"))
+  {
+    maxVel = _sdf->Get<double>("max_velocity");
+    hasVelocityLimits = true;
+  }
+  if (_sdf->HasElement("min_acceleration"))
+  {
+    minAccel = _sdf->Get<double>("min_acceleration");
+    hasAccelerationLimits = true;
+  }
+  if (_sdf->HasElement("max_acceleration"))
+  {
+    maxAccel = _sdf->Get<double>("max_acceleration");
+    hasAccelerationLimits = true;
+  }
+  if (_sdf->HasElement("min_jerk"))
+  {
+    minJerk = _sdf->Get<double>("min_jerk");
+    hasJerkLimits = true;
+  }
+  if (_sdf->HasElement("max_jerk"))
+  {
+    maxJerk = _sdf->Get<double>("max_jerk");
+    hasJerkLimits = true;
+  }
+
+  // Instantiate the speed limiters.
+  this->dataPtr->limiterLin = std::make_unique<SpeedLimiter>(
+    hasVelocityLimits, hasAccelerationLimits, hasJerkLimits,
+    minVel, maxVel, minAccel, maxAccel, minJerk, maxJerk);
+
+  this->dataPtr->limiterAng = std::make_unique<SpeedLimiter>(
+    hasVelocityLimits, hasAccelerationLimits, hasJerkLimits,
+    minVel, maxVel, minAccel, maxAccel, minJerk, maxJerk);
+
+  double odomFreq = _sdf->Get<double>("odom_publish_frequency", 50).first;
+  if (odomFreq > 0)
+  {
+    std::chrono::duration<double> odomPer{1 / odomFreq};
+    this->dataPtr->odomPubPeriod =
+      std::chrono::duration_cast<std::chrono::steady_clock::duration>(odomPer);
+  }
+
+
+  // Subscribe to commands
+  std::vector<std::string> topics;
+  if (_sdf->HasElement("topic"))
+  {
+    topics.push_back(_sdf->Get<std::string>("topic"));
+  }
+  topics.push_back("/model/" + this->dataPtr->model.Name(_ecm) + "/cmd_vel");
+  auto topic = validTopic(topics);
```

Comment:
You should check if the output of validTopic is empty. https://ignitionrobotics.org/api/transport/9.0/classignition_1_1transport_1_1TopicUtils.html#a42548e276fce157d9794eb8a05bf07b3

here and below

Reply:
Thank you for looking at the code.  All requested changes have been made and checked in.  codecheck and tests have been re-run and pass.
https://github.com/ignitionrobotics/ign-gazebo/pull/618/commits/7d56600edb0dd7d9695ee207a4f98a177f8b414e#diff-0dcac5ee6e5c74605877efaf6a30d6c6b5b23ef9d867f238e7a4b5185f1c4de0R327-R331